application/src/main/java/org/togetherjava/tjbot/features/basic/SlashCommandEducator.java

```diff
@@ -35,7 +35,9 @@ public void onMessageReceived(MessageReceivedEvent event) {
 
         String content = event.getMessage().getContentRaw();
 
-        if (IS_MESSAGE_COMMAND.test(content) && content.length() < MAX_COMMAND_LENGTH) {
+        if (IS_MESSAGE_COMMAND.test(content) &&
+            content.length() < MAX_COMMAND_LENGTH &&
+            (!content.endsWith(";") && !content.endsWith(")"))) {
```

Comment:
the extra `(...)` is obsolete

Reply:
do this `!(content.endsWith(";") || content.endsWith(")"))`

Reply:
Disagree. The two checks are separate checks. To me, the checks read like:
* is message command?
* message small?
* message doesnt end with ;?
* message doesnt end with )?

That said, at this point the better approach would likely be to just add all those checks into the regex `IS_MESSAGE_COMMAND`.

Reply:
Thats the current pattern:
```java
    private static final Predicate<String> IS_MESSAGE_COMMAND = Pattern.compile("""
            [.!?] #Start of message command
            [a-zA-Z]{2,15} #Name of message command, e.g. 'close'
            .* #Rest of the message
""", Pattern.COMMENTS).asMatchPredicate();
```
Thats how to add this part:
```java
    private static final Predicate<String> IS_MESSAGE_COMMAND = Pattern.compile("""
            [.!?] #Start of message command
            [a-zA-Z]{2,15} #Name of message command, e.g. 'close'
            .*[^;)] #Rest of the message
""", Pattern.COMMENTS).asMatchPredicate();
```

Reply:
Hi guys! Thanks for the feedback.

I did see the regex - but felt that it was best left alone. I did initially  try to do regex for the test but I couldn't get it to work. But this change makes way more sense. I also was just following the instructions left for the issue. 

What do you mean by:

> the extra (...) is obsolete

Reply:
Thanks 👍 You wrote
```java
if (foo ... && (!bar && !baz))
               ^            ^
``` 
Which is the same as just writing
```java
if (foo ... && !bar && !baz)
```